compiler/env/OMRAotAdapter.cpp

```diff
@@ -0,0 +1,143 @@
+}
+
+
+void OMR::AotAdapter::storeAOTMethodAndDataInTheCache(const char* methodName){
+    TR::AOTMethodHeader* hdr =_methodNameToHeaderMap[methodName];
+    // _sharedCache->storeEntry(methodName,hdr->serialize(),hdr->sizeOfSerializedVersion());
+}
+
+TR::AOTMethodHeader* OMR::AotAdapter::loadAOTMethodAndDataFromTheCache(const char* methodName)
+{
+    TR::AOTMethodHeader* methodHeader = NULL;
+    void* cacheEntry;
+    // void* cacheEntry = reinterpret_cast<uint8_t*> (_sharedCache->loadEntry(methodName));
+    if (cacheEntry!=NULL){
+        methodHeader = new TR::AOTMethodHeader((uint8_t*)cacheEntry);
+        void* codeStart = methodHeader->compiledCodeStart;
+        _reloRuntime->registerLoadedSymbol(methodName, codeStart);
+        registerAOTMethodHeader(methodName,methodHeader);
+    }
+
+    return methodHeader;
+}
+void OMR::AotAdapter::registerAOTMethodHeader(std::string methodName,TR::AOTMethodHeader* header){
+    _methodNameToHeaderMap[methodName] = header;
+}
+
+TR::RelocationRuntime* OMR::AotAdapter::getRelocationRuntime(){
+    return _reloRuntime->self();
+}
+
+TR::AOTMethodHeader* OMR::AotAdapter::createAndRegisterAOTMethodHeader(const char* methodName, uint8_t* codeStart, uint32_t codeSize,
+						  uint8_t* dataStart, uint32_t dataSize){
+     TR::AOTMethodHeader* hdr = (TR::AOTMethodHeader*)new (TR::AOTMethodHeader) (codeStart,codeSize,dataStart,dataSize);
+     registerAOTMethodHeader(methodName,hdr );
+     return hdr;
+}
+
+void OMR::AotAdapter::relocateRegisteredMethod(const char* methodName)
+{
+    TR::AOTMethodHeader* hdr =  getRegisteredAOTMethodHeader(methodName);
+    _reloRuntime->self()->prepareRelocateAOTCodeAndData(hdr,hdr->compiledCodeStart);
+}
+
+ TR::AOTMethodHeader* OMR::AotAdapter::getRegisteredAOTMethodHeader(const char* methodName){
+    std::string method(methodName);
+    TR::AOTMethodHeader* result =_methodNameToHeaderMap[methodName];
+    if (result==NULL)
+        {
+        result = loadAOTMethodAndDataFromTheCache(methodName);
+        }
+    return result;
+ }
+void OMR::AotAdapter::storeHeaderForCompiledMethod(const char* methodName)
+{
+    std::string method(methodName);
+
+    if (_methodNameToHeaderMap[methodName] != NULL)
+    {
+       storeAOTMethodAndDataInTheCache(methodName);
+    } else {
+       std::cerr << "Method " << methodName << " not found!" << std::endl;
+    }
+}
+bool isMethodAllocatedAlready(void* pc){
+    return false;
+}
+
+void* OMR::AotAdapter::getMethodCode(const char* methodName)
+    {
+    TR::AOTMethodHeader* methodHeader = getRegisteredAOTMethodHeader(methodName);
+    if (methodHeader == NULL)
+        return NULL;
+    if (false == isMethodAllocatedAlready(methodHeader->compiledCodeStart))
+        {
+        int32_t numReserved;
+        TR::CodeCache *codeCache = _codeCacheManager->reserveCodeCache(false, methodHeader->compiledCodeSize, 0, &numReserved);
+        if(!codeCache)
+            {
+            return nullptr;
```

Comment:
`nullptr` is not supported on all OMR build compilers unfortunately.  Please use `NULL`.

Reply:
Hi, thank you, all nullptr's in that file are NULL now.